tools/document-pii/src/main.rs

```diff
@@ -0,0 +1,233 @@
+            Some(path) => {
+                if !path.join("Cargo.toml").exists() {
+                    anyhow::bail!("Please provide the path to a rust crate/workspace");
+                }
+                path
+            }
+            None => std::env::current_dir()?,
+        };
+
+        // Before we can iterate over the PII fields properly, we make a mapping between all
+        // paths to types and their AST node, and of all modules and the items in their scope.
+        let types_and_use_statements = {
+            let rust_file_paths = find_rs_files(&path);
+            AstItemCollector::collect(&rust_file_paths)?
+        };
+
+        let pii_types = match self.item.as_deref() {
+            // If user provides path to an item, find PII_fields under this item in particular.
+            Some(path) => types_and_use_statements.find_pii_fields_of_type(path, &self.pii_values),
+            // If no item is provided, find PII fields of all types in crate/workspace.
+            None => types_and_use_statements.find_pii_fields_of_all_types(&self.pii_values),
+        }?;
+
+        // Function also takes a string to replace unnamed fields, for now we just remove them.
+        let output_vec = get_pii_fields_output(pii_types, "".to_string());
+
+        match self.output {
+            Some(ref path) => self.write_pii(File::create(path)?, &output_vec)?,
+            None => self.write_pii(std::io::stdout(), &output_vec)?,
+        }
+
+        Ok(())
+    }
+}
+
+/// Represent the PII fields in a format that will be used in the final output.
+fn get_pii_fields_output(
+    pii_types: BTreeSet<Vec<TypeAndField>>,
+    unnamed_replace: String,
+) -> Vec<String> {
+    let mut output_vec = vec![];
+    for pii in pii_types {
+        let mut output = String::new();
+        output.push_str(&pii[0].qualified_type_name);
+
+        for path in pii {
+            output.push_str(&format!(".{}", path.field_ident));
+        }
+
+        output = output.replace("{{Unnamed}}.", &unnamed_replace);
+        output_vec.push(output);
+    }
+    output_vec.sort();
+    output_vec
+}
+
+fn print_error(error: &anyhow::Error) {
+    eprintln!("Error: {error}");
+
+    let mut cause = error.source();
+    while let Some(ref e) = cause {
+        eprintln!("  caused by: {e}");
+        cause = e.source();
+    }
+}
+
+fn main() {
+    let cli = Cli::parse();
+
+    match cli.run() {
+        Ok(()) => (),
+        Err(error) => {
+            print_error(&error);
+            std::process::exit(1);
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use path_slash::PathBufExt;
+
+    use crate::item_collector::{AstItemCollector, TypesAndScopedPaths};
+
+    use super::*;
+
+    const RUST_TEST_CRATE: &str = "../../tests/test_pii_docs";
+
+    fn get_types_and_use_statements() -> TypesAndScopedPaths {
+        let rust_crate = PathBuf::from_slash(RUST_TEST_CRATE);
+        let rust_file_paths = find_rs_files(&rust_crate);
```

Comment:
Since the test crate is only for testing purposes, the number and name of files won't change often. What do you think about hardcoding the list of files, to not rely on `find_rs_files`?

Reply:
I could do that, but isn't it better to verify that the `find_rs_files` works as expected, as is done now?

Reply:
I made now a specific test to see that it got all the files